cosmoz-tree.html

```diff
@@ -18,60 +25,131 @@
 		},
 
-		getChildren: function (node) {
+		/**
+		 * Should returns a string which describes the path of a node (found by a node's property and value).
+		 * @return {string} e.g. home/computer/desktop
+		 * @param {string} propertyName (The name of the property the match should be based on. e.g. "name")
+		 * @param {string} propertyValue (The value of the property the match should be based on. e.g. "Peter")
+		 * @param {string} pathProperty (The property of a node on which the path should be build on. e.g "location" if node = {"location": "home"})
+		 * @param {string} pathSeparator (The string the path should get separated with.)
+		 * @param {string} separatorSign (The string which separates the path. e.g ".")
+		 * @param {string} childProperty (The string which describes the child property of a node.)
+		*/
+		getPathStringByProperty: function (propertyName, propertyValue, pathProperty, pathSeparator, separatorSign, childProperty) {
 			throw new Error('Must be implemented in derived object');
 		},
 
-		getProperty: function (node, propertyName) {
+		/** TODO (jalal): Drop this method
+		 * Should return the children of a node in form of an array.
+		 * @return {array}
+		 * @param {node} node (The node of which the children should be returned of.)
+		 * @param {string} childProperty (The string which describes the child property of a node.)
+		*/
+		getChildren: function (node, childProperty) {
 			throw new Error('Must be implemented in derived object');
 		},
 
-		getPathString: function (node, textPropertyName, separator) {
-			var path = this.getPath(node);
-			return path.map(function (n) {
-				return this.getProperty(n, textPropertyName);
-			}, this).join(separator);
+		/** TODO (jalal): Drop this method
+		 * Should return the value of a node's property.
+		 * @return {array}
+		 * @param {node} node (The node of which the property value should be returned of.)
+		 * @param {string} propertyName (The name of the property. e.g. "name")
+		*/
+		getProperty: function (node, propertyName) {
+			throw new Error('Must be implemented in derived object');
 		},
 
-		getPathStringByProperty: function (propertyName, propertyValue, textPropertyName, separator) {
-			var node = this.getNodeByProperty(propertyName, propertyValue);
-			if (node !== undefined) {
-				return this.getPathString(node, textPropertyName, separator);
+		/**
+		 * Checks if a node matches the search criteria.
+		 * @return {boolean}
+		 * @param {node} node (The node the check should be based on.)
+		 * @param {string} propertyName (The name of the property the match should be based on. e.g. "name")
+		 * @param {string} propertyValue (The value of the property the match should be based on. e.g. "Peter")
+		 * @param {boolean} exact (If the search should be executed exact or flaw. true wouldn't match "Pet")
+		*/
+		nodeConformsSearch: function (node, propertyName, propertyValue, exact) {
+			var property = this.getProperty(node, propertyName);
+
```

Comment:
You could also do

```
if (!property) {
  return;
}
```
to be sure that the property exists for the remainder of the function.
Makes it easier to maintain.